hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckpointServlet.java

```diff
@@ -175,23 +188,94 @@ public static Set<Path> normalizeExcludeList(List<String> toExcludeList,
     return paths;
   }
 
+  @Override
+  // Pauses rocksdb compaction threads while creating copies of
+  // compaction logs and hard links of sst backups.
+  public DBCheckpoint getCheckpoint(Path tmpdir, boolean flush)
+      throws IOException {
+    DBCheckpoint checkpoint;
+
+    // make tmp directories to contain the copies
+    RocksDBCheckpointDiffer differ = getDbStore().getRocksDBCheckpointDiffer();
+    DirectoryData sstBackupDir = new DirectoryData(tmpdir,
+        differ.getSSTBackupDir());
+    DirectoryData compactionLogDir = new DirectoryData(tmpdir,
+        differ.getCompactionLogDir());
+    if (!sstBackupDir.getTmpDir().mkdirs()) {
+      throw new IOException("mkdirs failed: " + sstBackupDir.getTmpDir());
+    }
+    if (!compactionLogDir.getTmpDir().mkdirs()) {
+      throw new IOException("mkdirs failed: " + compactionLogDir.getTmpDir());
+    }
+
+    long startTime = System.currentTimeMillis();
+    long pauseCounter = PAUSE_COUNTER.incrementAndGet();
+
+    // Pause compactions, Copy/link files and get checkpoint.
+    try {
+      LOG.info("Compaction pausing {} started.", pauseCounter);
+      differ.incrementTarballRequestCount();
+      FileUtils.copyDirectory(compactionLogDir.getDir(),
+          compactionLogDir.getTmpDir());
+      OmSnapshotUtils.linkFiles(sstBackupDir.getDir(),
+          sstBackupDir.getTmpDir());
+      checkpoint = getDbStore().getCheckpoint(flush);
+    } finally {
+      // Unpause the compaction threads.
+      synchronized (getDbStore().getRocksDBCheckpointDiffer()) {
+        differ.decrementTarballRequestCount();
+        differ.notifyAll();
+        long elapsedTime = System.currentTimeMillis() - startTime;
+        LOG.info("Compaction pausing {} ended. Elapsed ms: {}",
+            pauseCounter, elapsedTime);
+      }
+    }
+
+    return checkpoint;
+  }
+
+
+  // Convenience class for keeping track of the tmp dir's.
+  private static class DirectoryData {
+    private final File dir;
```

Comment:
nit: may be `sourceDir` or `srcDir` instead of just dir.

Reply:
unfortunately srcdir has a couple of different meanings.  in other places in the code, this is the dest dir, not the source dir.

how about I change it to originalDir, instead of srcDir?  is that ok?

Reply:
`originalDir` is fine by me.

Reply:
done